homeassistant/components/nuki/__init__.py

```diff
@@ -39,11 +39,23 @@ def _get_bridge_devices(bridge: NukiBridge) -> tuple[list[NukiLock], list[NukiOp
     return bridge.locks, bridge.openers
 
 
-def _update_devices(devices: list[NukiDevice]) -> None:
+def _update_devices(hass: HomeAssistant, devices: list[NukiDevice]) -> None:
     for device in devices:
         for level in (False, True):
             try:
-                device.update(level)
+                if isinstance(device, NukiOpener):
+                    last_ring_action_state = device.ring_action_state
+
+                    device.update(level)
+
+                    if not last_ring_action_state and device.ring_action_state:
+                        event_data = {
+                            "entity_id": device.entity.entity_id,
+                            "type": "ring",
+                        }
+                        hass.bus.async_fire("nuki_event", event_data)
```

Comment:
We're inside the thread pool, ie not inside the event loop. So we can't use the async interface.

```suggestion
                        hass.bus.fire("nuki_event", event_data)
```